Sources/Controllers/RoutePlanning/OASaveTrackViewController.mm

```diff
@@ -117,6 +117,23 @@ - (NSString *) getDisplayingFolderName:(NSString *)filePath
 - (void)viewWillAppear:(BOOL)animated
 {
     [super viewWillAppear:animated];
+    
+    [self.navigationController setNavigationBarHidden:NO animated:YES];
```

Comment:
The code for styling the navigation bar should be placed in a separate method, configureNavigationBar